### ISSUE TITLE
Run tests in a unique temp directory?

### ISSUE THREAD

**Issue by mrjbq7** [factor/factor#1605]
Maybe instead of doing tests like this:

```factor
[
    [
        { "hello" } [
            "hello" "foo" utf8 set-file-contents
            "foo" utf8 file-contents
        ] unit-test
    ] cleanup-unique-directory
] with-temp-directory
```

Instead, we should modify ``test`` to either run each ``test-vocab`` in it's own unique temp directory (although that might create too many unique temp directories since most don't need them), or have the test runner create a unique temp directory then run all the requested tests in that directory, then cleanup...

Thoughts @erg?

**Comment by erg** [comment]
Maybe some combinators ``with-test-file`` or ``with-test-directory`` would help?

**Comment by mrjbq7** [comment]
Okay, lets do that for now.

Fixed in dd3189ecb6b20fcb878a5534e7d88ada40728005.
